options/posix/generic/pthread-stubs.cpp

```diff
@@ -887,42 +887,54 @@ int pthread_mutex_consistent(pthread_mutex_t *) {
 // pthread_condattr and pthread_cond functions.
 // ----------------------------------------------------------------------------
 
-int pthread_condattr_init(pthread_condattr_t *) {
-	SCOPE_TRACE();
-
-	mlibc::infoLogger() << "mlibc: pthread_condattr_init() is not implemented correctly" << frg::endlog;
+int pthread_condattr_init(pthread_condattr_t *attr) {
+	attr->__mlibc_pshared = PTHREAD_PROCESS_PRIVATE;
+	attr->__mlibc_clock = CLOCK_REALTIME;
 	return 0;
 }
-int pthread_condattr_destroy(pthread_condattr_t *) {
-	SCOPE_TRACE();
 
-	mlibc::infoLogger() << "mlibc: pthread_condattr_destroy() is not implemented correctly" << frg::endlog;
+int pthread_condattr_destroy(pthread_condattr_t *attr) {
+	memset(attr, 0, sizeof(*attr));
 	return 0;
 }
 
-int pthread_condattr_getclock(const pthread_condattr_t *__restrict, clockid_t *__restrict) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_condattr_getclock(const pthread_condattr_t *__restrict attr,
+		clockid_t *__restrict clock) {
+	*clock = attr->__mlibc_clock;
+	return 0;
 }
-int pthread_condattr_setclock(pthread_condattr_t *, clockid_t) {
-	SCOPE_TRACE();
 
-	mlibc::infoLogger() << "mlibc: pthread_condattr_setclock() is not implemented correctly" << frg::endlog;
+int pthread_condattr_setclock(pthread_condattr_t *attr, clockid_t clock) {
+	if (clock != CLOCK_REALTIME && clock != CLOCK_MONOTONIC)
+		return EINVAL;
+
+	attr->__mlibc_clock = clock;
 	return 0;
 }
 
-int pthread_condattr_getpshared(const pthread_condattr_t *__restrict, int *__restrict) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_condattr_getpshared(const pthread_condattr_t *__restrict attr,
+		int *__restrict pshared) {
+	*pshared = attr->__mlibc_pshared;
+	return 0;
 }
-int pthread_condattr_setpshared(pthread_condattr_t *, int) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+
+int pthread_condattr_setpshared(pthread_condattr_t *attr, int pshared) {
+	if (pshared != PTHREAD_PROCESS_PRIVATE && pshared != PTHREAD_PROCESS_SHARED)
+		return EINVAL;
+
+	attr->__mlibc_pshared = pshared;
+	return 0;
 }
 
-int pthread_cond_init(pthread_cond_t *__restrict cond, const pthread_condattr_t *__restrict) {
+int pthread_cond_init(pthread_cond_t *__restrict cond, const pthread_condattr_t *__restrict attr) {
 	SCOPE_TRACE();
 
+	auto clock = attr ? attr->__mlibc_clock : CLOCK_REALTIME;
+	auto pshared = attr ? attr->__mlibc_pshared : PTHREAD_PROCESS_PRIVATE;
+
+	// Since we don't implement these yet, set a flag to error later.
+	cond->__mlibc_flags = clock || pshared;
```

Comment:
How is this going to work if we implement it properly? Can we compress the clock into flags? AFAICT that's not possible since the user might pass the result of `pthread_getcpuclockid()` or similar functions.

Reply:
Hmm, setclock() isn't allowed to accept a CPU-time clock, so that wouldn't work. But perhaps we should add a `clockid_t` field on `pthread_cond_t` just to be safe

Reply:
Yeah, I think that a single unsigned int should be sufficient here. Even if we need to store more clockid_t values, we'd probably fine since we only need 1 bit to store whether the condition variable is process-shared or not.

Reply:
I've added a clockid_t field on cond_t now.